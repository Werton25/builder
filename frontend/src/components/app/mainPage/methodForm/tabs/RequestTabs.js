import React from 'react';
import PropTypes from 'prop-types';
import Editor from "../editor/Editor";
import {connect} from 'react-redux';
import {Tabs, TabLink, TabContent} from 'react-tabs-redux';
import {getCurrentRequest, getDefaultRequestSchema} from "../../../../../selectors/requests";
import {isFullScreenMode} from "../../../../../selectors/editor";
import {getSelectedTab} from "../../../../../selectors/tabs";
import {isDefaultRequest} from "../../../../../selectors/select";
import {changeSelected} from "../../../../../actions/tabs";
import {toggleFullScreenMode} from "../../../../../actions/editor";
import {editRequest} from '../../../../../actions/requests';
import './Tabs.css';
import {getCurrentUri} from "../../../../../selectors/router";

export class BaseRequestTabs extends React.Component {
  render() {
    const {
      currentRequest, currentUri, isDefaultRequest,
      requestSchema, changeSelected, toggleFullScreenMode,
      isFullScreenMode, selectedTab, editRequest
    } = this.props;
    return (
      <Tabs className='tabs tabs-1' selectedTab={selectedTab}
            handleSelect={(tab) => changeSelected(tab, currentUri, currentRequest.id)}>
        <div className='tab-links'>
          <TabLink to='value'>Запрос</TabLink>
          <TabLink to='schema'>Описание параметров</TabLink>
        </div>
        <div className='content'>
          <TabContent for='value'>
            <Editor value={currentRequest.value}
                    onChange={(value) => editRequest({id: currentRequest.id, uri: currentUri, value})}
                    onToggleFullScreenMode={toggleFullScreenMode}
                    options={{
                      lint: true,
                      autoCloseBrackets: true,
                      readOnly: isDefaultRequest,
                      fullScreen: isFullScreenMode,
                    }}/>
          </TabContent>
          <TabContent for='schema'>
            <Editor value={requestSchema}
                    options={{readOnly: true}}/>
          </TabContent>
        </div>
      </Tabs>
    );
  }
}

BaseRequestTabs.propTypes = {
  currentUri: PropTypes.string.isRequired,
  currentRequest: PropTypes.object.isRequired,
  isDefaultRequest: PropTypes.bool.isRequired,
  requestSchema: PropTypes.string.isRequired,
  isFullScreenMode: PropTypes.bool.isRequired,
  selectedTab: PropTypes.string,
  changeSelected: PropTypes.func.isRequired,
  editRequest: PropTypes.func.isRequired,
  toggleFullScreenMode: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  currentUri: getCurrentUri(state),
  currentRequest: getCurrentRequest(state),
  isDefaultRequest: isDefaultRequest(state),
  requestSchema: getDefaultRequestSchema(state),
  isFullScreenMode: isFullScreenMode(state),
  selectedTab: getSelectedTab(state),
});

const mapDispatchToProps = {
  changeSelected,
  editRequest,
  toggleFullScreenMode
};

export default connect(mapStateToProps, mapDispatchToProps)(BaseRequestTabs);
