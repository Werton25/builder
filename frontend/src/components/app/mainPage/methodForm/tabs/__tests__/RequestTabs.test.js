import React from "react";
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import RequestTabs from "../RequestTabs";

jest.mock('connected-react-router');
jest.mock('../../editor/Editor');
jest.mock('../../../../../../selectors/editor', () => ({isFullScreenMode: () => false}));
jest.mock('../../../../../../selectors/router', () => ({getCurrentUri: () => '/service/method'}));
jest.mock('../../../../../../selectors/tabs', () => ({getSelectedTab: () => 'value'}));
jest.mock('../../../../../../selectors/select', () => ({isDefaultRequest: () => false}));
jest.mock('../../../../../../selectors/requests', () => ({
  getDefaultRequestSchema: () => '{}',
  getCurrentRequest: () => ({
    id: '1',
    name: 'test',
    value: '{}'
  })
}));
const store = configureStore()({});

describe('RequestTabs tests', () => {
  it('RequestTabs renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RequestTabs/>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
