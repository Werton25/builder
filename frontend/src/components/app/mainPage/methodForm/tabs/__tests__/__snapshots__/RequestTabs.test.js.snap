// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestTabs tests RequestTabs renders correctly 1`] = `
<div
  className="tabs tabs-1"
>
  <div
    className="tab-links"
  >
    <div
      className="tab-link tab-link-active"
      onClick={[Function]}
      onKeyPress={[Function]}
      style={
        Object {
          "fontWeight": "bold",
        }
      }
      tabIndex="0"
    >
      Запрос
    </div>
    <div
      className="tab-link"
      onClick={[Function]}
      onKeyPress={[Function]}
      style={Object {}}
      tabIndex="0"
    >
      Описание параметров
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="tab-content tab-content-visible"
      style={Object {}}
    />
    <div
      className="tab-content"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</div>
`;
